packages/pages/src/common/src/parsers/templateParser.test.ts

```diff
@@ -0,0 +1,150 @@
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).toContain(`const foo = 5;`);
+    expect(templateParser.fileContents).toContain(`export default foo`);
+  });
+
+  it("does not return unnecessary contents ", () => {
+    const sourceParser = createParser(
+      `const foo = 5; const bar = 7; export default foo`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).not.toContain(`const bar = 7;`);
+    expect(templateParser.fileContents).toContain(`const foo = 5;`);
+    expect(templateParser.fileContents).toContain(`export default foo`);
+  });
+
+  it("correctly returns child expressions", () => {
+    const sourceParser = createParser(
+      `const foo = bar + 5; const bar = 7; export default foo`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).toContain(`const foo = bar + 5;`);
+    expect(templateParser.fileContents).toContain(`const bar = 7;`);
+    expect(templateParser.fileContents).toContain(`export default foo`);
+  });
+
+  it("correctly returns imports", () => {
+    const sourceParser = createParser(
+      `import { Template, TemplateRenderProps } from "@yext/pages"; 
+      const foo: Template<TemplateRenderProps> = () => { return (<></>); };
+      export default foo;`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).toContain(
+      `import { Template, TemplateRenderProps } from "@yext/pages";`
+    );
+  });
+
+  it("does not return unnecessary imports", () => {
+    const sourceParser = createParser(
+      `import { Template, TemplateRenderProps } from "@yext/pages";`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).not.toContain(
+      `import { Template, TemplateRenderProps } from "@yext/pages";`
+    );
+  });
+
+  it("does not return tsdoc comments", () => {
+    const sourceParser = createParser(
+      `/**
+      * mock tsdoc comments
+      */
+      const foo = 5;
+      export default foo`
+    );
+    const testParser = createParser(``);
+    const templateParser = new TemplateParser(
+      sourceParser
+    ).makeClientTemplateFromSfp(testParser);
+    expect(templateParser.fileContents).not.toContain(` mock tsdoc comments`);
+  });
+});
+
+describe("complex makeClientTemplate usages", () => {
```

Comment:
Cannot directly compare expected results to a file due to the imports being absolute paths after being processed.

Reply:
No longer true. Now uses relative paths